content/en/blog/_posts/2024-02-29-cri-o-seccomp-oci-artifacts.md

```diff
@@ -0,0 +1,421 @@
+pod "pod" deleted
+
+> kubectl apply -f pod.yaml
+pod/pod created
+```
+
+The CRI-O logs will now indicate that the runtime pulled the artifact:
+
+```
+WARN[…] Allowed annotations are specified for workload [io.kubernetes.cri-o.seccompProfile]
+INFO[…] Found container specific seccomp profile annotation: io.kubernetes.cri-o.seccompProfile/container=quay.io/crio/seccomp:v1  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+INFO[…] Pulling OCI artifact from ref: quay.io/crio/seccomp:v1  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+INFO[…] Trying to read profile from: /var/lib/containers/storage/overlay/c15f3eb1efd9abe0ae973e050c374988bba52c45a53f5c3cd80419a865013fa6/merged/seccomp.json  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+INFO[…] Retrieved OCI artifact seccomp profile of len: 18853  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+```
+
+And the container is finally using the profile:
+
+```
+> export CONTAINER_ID=$(sudo crictl ps --name container -q)
+
+> sudo crictl inspect $CONTAINER_ID | jq .info.runtimeSpec.linux.seccomp | head
+{
+  "defaultAction": "SCMP_ACT_ERRNO",
+  "defaultErrnoRet": 38,
+  "architectures": [
+    "SCMP_ARCH_X86_64",
+    "SCMP_ARCH_X86",
+    "SCMP_ARCH_X32"
+  ],
+  "syscalls": [
+    {
+```
+
+The same would work for every container in the pod, if users omit the
+`/container` suffix, for example:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+  annotations:
+    io.kubernetes.cri-o.seccompProfile: quay.io/crio/seccomp:v1
+spec:
+  containers:
+    - name: container
+      image: nginx:1.25.3
+```
+
+{{< note >}}
+CRI-O has an option called `seccomp_use_default_when_empty` (config file) /
+`--seccomp-use-default-when-empty` (CLI), which is enabled by default and sets
+the `RuntimeDefault` profile if not specified in the pod spec. For testing the
+new annotation, either disable it or lookout for the log line: _"Found container
+specific seccomp profile annotation"_.
+{{< /note >}}
+
+### Using the annotation for a container image
+
+While specifying seccomp profiles as OCI artifacts on certain workloads is a
+cool feature, the majority of end users would like to link seccomp profiles to
+actual container images. This can be done by using the same annotation, but now
+at the container image itself. For example, [Podman](https://podman.io) can be
+used to add the annotation directly on image build:
+
+```
+> podman build \
+    --annotation io.kubernetes.cri-o.seccompProfile=quay.io/crio/seccomp:v1 \
+    -t quay.io/crio/nginx-seccomp .
+```
+
+The pushed image then contains the annotation:
+
+```
+> skopeo inspect --raw docker://quay.io/crio/nginx-seccomp:latest |
+    jq '.annotations."io.kubernetes.cri-o.seccompProfile"'
+"quay.io/crio/seccomp:v1"
+```
+
+When we now use that image in our CRI-O test pod definition:
```

Comment:
```suggestion
If I now use that image in an CRI-O test pod definition:
```

We [avoid using “we”](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we); there's an exception for using it to refer to the multiple authors of a blog article, but that doesn't apply here.